classes/travel/TravelManager.php

```diff
@@ -403,24 +435,266 @@ public function fetchCurrentMapLocations(): array {
+            }
+        }
+        $result = $this->system->db->query("SELECT * FROM `caravans`");
+        $result = $this->system->db->fetch_all($result);
+        foreach ($result as $row) {
+            // if travel time is set then only display if active
+            if (!empty($row['travel_time'])) {
+                if ($row['travel_time'] + ($row['start_time'] * 1000) + Patrol::DESTINATION_BUFFER_MS > (time() * 1000)) {
+                    $patrol = new Patrol($row, "caravan");
+                    $patrol->setLocation($this->system);
+                    $patrol->setAlignment($this->user);
+                    if ($this->user->location->distanceDifference(new TravelCoords($patrol->current_x, $patrol->current_y, $patrol->map_id)) <= $this->user->scout_range) {
+                        $return_arr[] = $patrol;
+                    }
+                }
+            } else {
+                $patrol = new Patrol($row, "caravan");
+                $patrol->setLocation($this->system);
+                $patrol->setAlignment($this->user);
+                if ($this->user->location->distanceDifference(new TravelCoords($patrol->current_x, $patrol->current_y, $patrol->map_id)) <= $this->user->scout_range) {
+                    $return_arr[] = $patrol;
+                }
+            }
+        }
+
+        return $return_arr;
+    }
+
+    /**
+     * @return MapObjective[]
+     */
+    #[Trace]
+    public function fetchMapObjectives(): array
+    {
         $objectives = [];
+
+        // Get mission objectives
         if ($this->user->mission_id > 0) {
             if ($this->user->mission_stage['action_type'] == 'travel') {
                 $mission_result = $this->system->db->query("SELECT `name` FROM `missions` WHERE `mission_id` = '{$this->user->mission_id}' LIMIT 1");
                 $mission_location = TravelCoords::fromDbString($this->user->mission_stage['action_data']);
-                $objectives[] = new MapObjectiveLocation(
+                $objectives[] = new MapObjective(
+                    id: null,
                     name: $this->system->db->fetch($mission_result)['name'],
                     map_id: $mission_location->map_id,
                     x: $mission_location->x,
                     y: $mission_location->y,
-                    image: "/images/v2/icons/anbutracking.png",
+                    image: "/images/v2/icons/reticle.png",
+                    objective_type: "target",
                 );
             }
             if ($this->user->mission_stage['action_type'] == 'search') {
                 $mission_result = $this->system->db->query("SELECT `name` FROM `missions` WHERE `mission_id` = '{$this->user->mission_id}' LIMIT 1");
                 $mission_location = TravelCoords::fromDbString($this->user->mission_stage['last_location']);
-                $objectives[] = new MapObjectiveLocation(
+                $objectives[] = new MapObjective(
+                    id: null,
```

Comment:
Good catch!
MapObjectives are basically the way to pass any additional content to display on the map. So before that was mission objective, events, and regions. Since the IDs aren't necessarily coming from a DB primary key we can start the keys at specific values.

The player's mission target can always be set to 1 for example, event IDs can always start at 1000, etc. That's essentially how they worked before. Just to make sure we can have consistent IDs for React to update without risk of overlapping.